packages/sdk/src/syncedStream.ts

```diff
@@ -21,31 +44,91 @@ export class SyncedStream extends Stream implements ISyncedStream {
+        this.logError = dlogError('csb:syncedStream').extend(userId)
         this.persistenceStore = persistenceStore
+        this.metrics = {
+            initializationTime: 0,
+            lastAccessTime: Date.now(),
+            retryCount: 0,
+            memoryUsage: 0,
+            successfulOperations: 0,
+            failedOperations: 0,
+        }
     }
 
-    async initializeFromPersistence(persistedData?: LoadedStream): Promise<boolean> {
-        const loadedStream =
-            persistedData ?? (await this.persistenceStore.loadStream(this.streamId))
-        if (!loadedStream) {
-            this.log('No persisted data found for stream', this.streamId, persistedData)
-            return false
+    private calculateMemoryUsage(): number {
+        const streamSize = {
+            streamId: this.streamId,
+            userId: this.userId,
+            events: this.view.events.size,
+            timeline: this.view.timeline.length,
         }
+        return JSON.stringify(streamSize).length / 1024 // KB
+    }
+
+    async initializeFromPersistence(persistedData?: LoadedStream): Promise<boolean> {
+        const startTime = performance.now()
+        this.metrics.lastAccessTime = Date.now()
+
         try {
-            super.initialize(
-                loadedStream.persistedSyncedStream.syncCookie,
-                loadedStream.persistedSyncedStream.minipoolEvents,
-                loadedStream.snapshot,
-                loadedStream.miniblocks,
-                loadedStream.prependedMiniblocks,
-                loadedStream.miniblocks[0].header.prevSnapshotMiniblockNum,
-                loadedStream.cleartexts,
-            )
-        } catch (e) {
-            this.log('Error initializing from persistence', this.streamId, e)
+            const loadedStream =
+                persistedData ?? (await this.persistenceStore.loadStream(this.streamId))
+            if (!loadedStream) {
+                this.log('No persisted data found for stream', this.streamId, persistedData)
+                return false
+            }
+
+            await Promise.race([
+                Promise.resolve().then(() => {
+                    return super.initialize(
+                        loadedStream.persistedSyncedStream.syncCookie,
+                        loadedStream.persistedSyncedStream.minipoolEvents,
+                        loadedStream.snapshot,
+                        loadedStream.miniblocks,
+                        loadedStream.prependedMiniblocks,
+                        loadedStream.miniblocks[0].header.prevSnapshotMiniblockNum,
+                        loadedStream.cleartexts,
+                    )
+                }),
+                new Promise((_, reject) =>
+                    setTimeout(
+                        () => reject(new Error('Initialization timeout')),
+                        INITIALIZATION_TIMEOUT,
+                    ),
+                ),
```

Comment:
It never takes 30 seconds to load a stream from indexdb. It takes about 30 to 300 milliseconds. This code isn't doing anything